cmd/README.md

```diff
@@ -0,0 +1,738 @@
+
+``` json
+[{"inputs":[],"name":"sayHello","outputs":[{"internalType":"string","name":"result","type":"string"}],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"string","name":"recipient","type":"string"}],"name":"setGreeting","outputs":[],"stateMutability":"nonpayable","type":"function"}]
+```
+
+## Precompile tool
+
+The precompile tool can take in 4 arguments. 
+
+
+### `--abi` (string):
+
+It needs an ABI input so it can bind it to the precompile template.
+
+`--abi ./contract-examples/contracts/contract-abis/IHelloWorld.abi`
+
+### `--type `(string):
+
+It takes in a type which it uses as a struct name for the precompile. This is optional 
+and will default to the abi name. 
+
+`--type HelloWorld`
+
+### `--pkg` (string):
+
+It also takes in a pkg, which is the package name to generate the precompile into. 
+This is optional and it defaults to "precompile".
+
+`--pkg precompile` 
+
+### `--out `(string):
+
+Finally it can take in an out which is the path and name of the output file of the generated precompile. 
+
+`--out ./precompile/hello_world.go`
+
+Currently it can only generate precompiles in Golang only.  
+
+
+## Generating the precompile 
+
+Now that we have an abi for the precompile gen tool to interact with. We can run the following command to generate our HelloWorld precompile!
+
+
+In the root of the repo run 
+``` 
+go run ./cmd/precompilegen/main.go --abi ./contract-examples/contracts/IHelloWorld.abi --type HelloWorld --pkg precompile --out ./precompile/hello_world.go
+```
+
+Wow! We just got a precompile template that's mostly complete located at `./precompile/hello_world.go`. Let's fill out the rest!
+
+The precompile gives us commented instructions on the first 25 lines of the autogenerated file. Let's look at the 10 steps and follow them step by step.
+
+``` go
+/* General guidelines for precompile development:
+1- Read the comment and set a suitable contract address in precompile/params.go. E.g:
+	HelloWorldAddress = common.HexToAddress("ASUITABLEHEXADDRESS")
+2- Set gas costs here
+3- It is recommended to only modify code in the highlighted areas marked with "CUSTOM CODE STARTS HERE". Modifying code outside of these areas should be done with caution and with a deep understanding of how these changes may impact the EVM.
+Typically, custom codes are required in only those areas.
+4- Add your upgradable config in params/precompile_config.go
+5- Add your precompile upgrade in params/config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create your genesis with your precompile enabled in tests/e2e/genesis/
+9- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+*/
+```
+
+## Step 1: Set Contract Address
+
+In `./precompile/params.go` let's set a precompile address. We can cut the
```

Comment:
Can we make this more clear by stating exactly where to find the original address that needs to be moved and then stating that it needs to be moved to in `./precompile/params.go` ?

Reply:
This starts out with `./precompile/params.go` but I think it would be more clear if we start with the location in the generated precompile file.

Reply:
fixed